mern-ecommerce/controllers/user.js

```diff
@@ -40,6 +41,20 @@ exports.update = (req, res) => {
   );
 };
 
+exports.productById = (req, res, next, id) => { 
+  Product.findById(id) 
+    .populate('products.product', 'name price')
+    .exec((err, product) => { 
+      if (err || !product) {
+        return res.status(400).json({
+          error: errorHandler(err),
```

Comment:
There is a typo in the `res.jsosn(orders)` line, it should be `res.json(orders)`.